src/analytics.js

```diff
@@ -436,12 +436,14 @@ window.addEventListener("beforeunload", async () => {
 });
 
 // Expose userland API.
-window.Altis.Analytics.updateEndpoint = Analytics.updateEndpoint;
-window.Altis.Analytics.record = (type, data = {}) =>
-	Analytics.record(
-		type,
-		{
-			attributes: getAttributes(data.attributes || {}),
-			metrics: getMetrics(data.metrics || {})
-		}
-	);
+if (!Object.keys(Data).length === 0) {
```

Comment:
The problem with this approach is that these functions will be undefined during a preview, however other code may still be trying to use them.

The goal here is to make these functions into no-op versions that do nothing eg. `window.Altis.Analytics.updateEndpoint = () => {}`

Reply:
Oh I figured they were using other parts of the file, not the actual data. Will look at this again.

Reply:
To clarify other code doesn't have access to anything in this file - just this bit at the end, we're extending the `window` object to expose these functions. Everything else is contained.